### PR TITLE
Memory Pair Game

### PR DESCRIPTION
**Pull request by vitaliykravchyk**
### Memory Pair Game

[Demo](https://vitaliykravchyk.github.io/memory-pair-game/) |
[Code base](https://github.com/vitaliykravchyk/memory-pair-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/vitaliykravchyk/memory-pair-game/js/main.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            winnerString.innerHTML = "Good job, You are winner!";
+
+            const restartButton = document.createElement('button');
+            restartButton.type = 'button';
+            restartButton.innerHTML = 'Restart';
+
+            itemsContainer.appendChild(winnerString);
+            itemsContainer.appendChild(restartButton);
+
+            document.querySelector('button').addEventListener('click', function(){
```
Comment (AMashoshyna) [inline submissions/vitaliykravchyk/memory-pair-game/js/main.js]: You can attach event listener before mounting the element or use `restartButton` variable to do it, no need to make a query.

File: submissions/vitaliykravchyk/memory-pair-game/js/main.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        
+        board.appendChild(fragment);
+        document.querySelector('main .container').appendChild(board);
+        document.querySelector('.board').addEventListener('click', workWithItems);
+    }
+    
+    function workWithItems(){
+        if(event.target.hasAttribute('data-item')){
+            //this checkings are blocking click on 3rd card
+            if(document.querySelectorAll('img').length <= 1){
```
[the remark this answers is not in the document]
Comment (AMashoshyna) [reply submissions/vitaliykravchyk/memory-pair-game/js/main.js]: Could you please explain?

[the remark this answers is not in the document]
Comment (vitaliykravchyk) [reply submissions/vitaliykravchyk/memory-pair-game/js/main.js]: I can't use first query for two checking's becasue between first and second query i add new img which change count of elements first query in relation to the second

[the remark this answers is not in the document]
Comment (AMashoshyna) [reply submissions/vitaliykravchyk/memory-pair-game/js/main.js]: Oh, I see. I missed, that function `createImg` changes the DOM. My bad!

**Comment by AMashoshyna** [comment]
Looks much better now! I will merge the PR.
As next step, you may want to think about splitting data and presentation. This is not required in this task, but it will be very important later.
Now you basically mix data and presentation. It's ok for small project, but hardly maintainable for bigger ones. What I mean is that you handle your logic (deciding game status) and presentation (change application appearance in one function. It is better to keep it separate. Also, you basically use DOM to store the data. To know which cards are available you need to query the DOM, which is expensive and inefficient. A better approach would be to keep data in an array (for example), work with this array and change the DOM whenever your data changes.
But as I said, your solution is fine at this stage!

**Pull request merged**